### PR TITLE
Search and replace must not do so in .git or .hg folders

### PR DESCRIPTION
**Pull request by destitutus**
see https://codio.myjetbrains.com/youtrack/issue/codio-9132

### CODE REVIEW

File: lib/replace.js [hunk 1]
```diff
@@ -116,8 +116,9 @@ module.exports = function (files, pattern, opts) {
     glob(locations, {mark: true, dot: true}, function (error, location) {
       if (error) return cb(error);
       location = _.uniq(location);
-      location = _.filter(location, function (item) {
-        return item.lastIndexOf('/') !== item.length - 1;
+      // Filter out all directories and .git .hg
+      location = _.filter(location, function (file) {
+        return file.lastIndexOf('/') !== file.length - 1 && file.indexOf('.git/') !== 0 && file.indexOf('.hg/') !== 0;
```
Comment (vvlevykin) [inline lib/replace.js]: I suggest using `_.startsWith`

Comment (destitutus) [reply lib/replace.js]: not supported in used lodash version

Comment (vvlevykin) [reply lib/replace.js]: Can we update the lodash version?

Comment (destitutus) [reply lib/replace.js]: We could, but it will involve a lot of testing and actions for rebuild and changes

Comment (vvlevykin) [reply lib/replace.js]: Ok, in that case please create two functions: startsWith and endsWith to improve readability

**Review by vvlevykin — APPROVED** [review]

**Pull request merged**